discovery/sync_manager.go

```diff
@@ -35,10 +35,15 @@ const (
 	DefaultMsgBytesBurst = 2 * 100 * 1_024
 
 	// DefaultMsgBytesPerSecond is the max bytes/s we'll permit for outgoing
-	// messages. Once tokens (bytes) have been taken from the bucket,
-	// they'll be refilled at this rate.
+	// messages for all the peers. Once tokens (bytes) have been taken from
+	// the bucket, they'll be refilled at this rate.
 	DefaultMsgBytesPerSecond = 100 * 1_024
 
+	// DefaultPeerMsgBytesPerSecond is the max bytes/s we'll permit for
+	// outgoing messages for a single peer. Once tokens (bytes) have been
+	// taken from the bucket, they'll be refilled at this rate.
+	DefaultPeerMsgBytesPerSecond = 50 * 1_024
```

Comment:
What about scaling this based on the total global value? So something like: a peer can only use up to 5% of the global limit?

Reply:
yeah make sense. Tho I think we may still need an absolute value, similar to how we handle the budget used in the sweeper. The idea here is to limit the per-peer bandwidth, assuming only one or two bad peers exist, like what we've seen from the logs. For large nodes, they can have hundreds of peers, so 5% may be too large. And for small nodes, they may have less than 10 peers, and 5% can be too conservative.